packages/mdc-snackbar/mdc-snackbar.scss

```diff
@@ -21,152 +21,183 @@
   }
 }
 
-.mdc-snackbar--action-on-bottom .mdc-snackbar__text {
-  margin-right: inherit;
+.mdc-snackbar__overlay {
+  position: absolute;
+  top: 0;
+  right: 0;
+  bottom: 0;
+  left: 0;
+  box-sizing: border-box;
+  z-index: -1;
 }
 
-.mdc-snackbar--action-on-bottom .mdc-snackbar__action-wrapper {
-  @include mdc-rtl-reflexive-box(margin, left, auto);
+.mdc-snackbar__label {
+  @include mdc-typography($mdc-snackbar-label-type-scale);
 
-  flex-direction: column;
-  justify-content: flex-start;
-  margin-top: -12px;
-  margin-bottom: 8px;
+  display: none;
+  flex-grow: 1;
+  box-sizing: border-box;
+  margin: 0;
+
+  // 14px top/bottom padding needed to make the height 48px.
+  padding: 14px 16px;
+
+  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-snackbar--open &,
+  .mdc-snackbar--closing & {
+    display: block;
+  }
+  // stylelint-enable plugin/selector-bem-pattern
 }
 
-.mdc-snackbar--multiline .mdc-snackbar__text {
-  height: 80px;
+// Used to prevent visual jank when announcing label text to screen readers.
+// See the `announce()` function in util.js for details.
+.mdc-snackbar__label::before {
+  display: inline;
+  content: attr(data-mdc-snackbar-label-text);
 }
 
-.mdc-snackbar__action-button {
-  @include mdc-typography(button);
-  @include mdc-theme-prop(color, secondary);
-
-  padding: 0;
-  transition: mdc-animation-exit-permanent(opacity, .3s);
-  border: none;
-  outline: none;
-  background-color: transparent;
-  opacity: 0;
-  user-select: none;
-  -webkit-appearance: none;
-  visibility: hidden;
+.mdc-snackbar__actions {
+  @include mdc-rtl-reflexive-property(margin, 0, $mdc-snackbar-padding);
 
-  &::-moz-focus-inner {
-    border: 0;
-  }
+  display: none;
+  flex-shrink: 0;
+  align-items: center;
+  box-sizing: border-box;
 
-  &:hover {
-    cursor: pointer;
+  // stylelint-disable plugin/selector-bem-pattern
+  .mdc-snackbar--open &,
+  .mdc-snackbar--closing & {
+    display: flex;
   }
+  // stylelint-enable plugin/selector-bem-pattern
 
-  &:not([aria-hidden]) {
-    visibility: inherit;
+  .mdc-snackbar--stacked & {
+    align-self: flex-end;
+    margin-bottom: $mdc-snackbar-padding;
   }
 }
 
-.mdc-snackbar--active .mdc-snackbar__text,
-.mdc-snackbar--active .mdc-snackbar__action-button:not([aria-hidden]) {
-  transition: mdc-animation-exit-permanent(opacity, .3s);
-  opacity: 1;
+.mdc-snackbar__action-button {
+  @include mdc-button-ink-color($mdc-snackbar-action-button-ink-color);
+  @include mdc-states($mdc-snackbar-action-button-ink-color);
+}
+
+.mdc-snackbar__action-icon {
+  @include mdc-icon-button-ink-color($mdc-snackbar-action-icon-ink-color);
 }
 
-// stylelint-disable plugin/selector-bem-pattern
-.mdc-snackbar--multiline.mdc-snackbar--action-on-bottom .mdc-snackbar__text {
-  margin: 0;
+// Two selectors are needed to increase specificity above `.material-icons`.
+// stylelint-disable-next-line selector-class-pattern
+.mdc-snackbar__action-icon.material-icons {
```

Comment:
Should this really be using material-icons in the selector? What would happen if someone used an image or svg? Should we do the same-class-twice trick instead?